api/src/test/java/org/pmiops/workbench/workspaces/resources/UserRecentResourceServiceTest.java

```diff
@@ -247,39 +246,58 @@ public void testUpdateNotebookAccessTime() {
   @Test
   public void testUserLimit() {
     DbWorkspace newWorkspace = new DbWorkspace();
-    newWorkspace.setWorkspaceId(2l);
+    newWorkspace.setWorkspaceId(2L);
     workspaceDao.save(newWorkspace);
+
+    long rowsCount = userRecentlyModifiedResourceDao.count();
+    assertThat(rowsCount).isEqualTo(0);
+
+    // record 3 recent resources
+
+    int initialEntryCount = 3;
+    String firstNotebook = "gs://someDirectory1/notebooks/notebook";
     userRecentResourceService.updateNotebookEntry(
-        newWorkspace.getWorkspaceId(), user.getUserId(), "gs://someDirectory1/notebooks/notebook");
+        newWorkspace.getWorkspaceId(), user.getUserId(), firstNotebook);
+
     CLOCK.increment(2000);
-    userRecentResourceService.updateNotebookEntry(2l, user.getUserId(), "notebooks");
+
+    userRecentResourceService.updateNotebookEntry(
+        newWorkspace.getWorkspaceId(), user.getUserId(), "notebooks");
     userRecentResourceService.updateCohortEntry(
         newWorkspace.getWorkspaceId(), user.getUserId(), cohort.getCohortId());
-    int count = UserRecentResourceService.USER_ENTRY_COUNT - 3;
-    while (count-- >= 0) {
+
+    // record enough recent resources to fill up the table
+
+    int toAdd = UserRecentResourceService.USER_ENTRY_COUNT - initialEntryCount;
+    while (toAdd-- > 0) {
```

Comment:
changed `>=` to `>` to clarify this a bit: now this loop fills up to the limit exactly instead of going over.